config/initializers/mime_types.rb

```diff
@@ -3,3 +3,4 @@
 # Add new mime types for use in respond_to blocks:
 # Mime::Type.register "text/richtext", :rtf
 # Mime::Type.register_alias "text/html", :iphone
+Mime::Type.register "application/x-gzip", :xml_gz, [], ["xml.gz"]
```

Comment:
The difference is that mime type applies to content being transferred and transfer encoding only applies to the transfer, i.e. the sitemap.xml is delivered over the wire as gzipped but the client delivers it to requesting app as xml. With application/x-gzip the content itself is gzipped so if you serve a sitemap.xml.gz file then that's appropriate mime type. I assume we only serve the xml and not a .gz file containing the xml? In that case application/xml is the correct mime type.